### DISCUSSION TITLE
Is it possible to stop a STARTING run via graphql

### DISCUSSION THREAD

**Discussion by gregd33** [dagster-io/dagster#4517] [Q&A] [answered]
I have noticed some issues with some runs getting stalled in a STARTING (I think it is a k8s issue). Naturally I will want to fix/address that but in the meantime, is there a way to use graphql to stop a run that is stuck in STARTING? 

It seems that it is not supported via TerminatePipeline (Run xxxxx could not be terminated due to having status STARTING.)

**Comment by gibsondan** [comment, marked as answer]
Hi, there's an optional argument to terminatePipelineExecution that can be used to 'force terminate' the run, moving it into a FAILURE state without necessarily cleaning up all the underlying resources that it spun up (for example, the k8s pod). This can be used to terminate runs that are stuck in STARTING.

sample mutation:
```
mutation($runId:ID!, $terminatePolicy:TerminatePipelinePolicy) {
  terminatePipelineExecution(runId:$runId, terminatePolicy:$terminatePolicy) {
    __typename
  }
}

with 

{"terminatePolicy": "MARK_AS_CANCELED_IMMEDIATELY", "runId": "your-run-id-here"}

```

Just filed an issue to include that in the documentation for the GraphQL API.
